Middleware with pre-v1 peerDependencies installs old version of them

I started using Valibot validator middleware and noticed that the installed Valibot is quite old. There are 22 releases between the installed (0.13.1) and the latest (0.30.0) versions.

This is because peer deps are declared with a caret notation and `^0.13.1` doesn't match 0.14.0 and later.

https://github.com/honojs/middleware/blob/c56b22f15ad9203d46b15440bd7764e45d244b99/packages/valibot-validator/package.json#L29-L32

This is particularly problematic when a middleware is installed in a project where its peer deps are already used. An `ERESOLVE` error is issued and you fail to install the middleware:

```console
$ npm install valibot
$ npm install @hono/valibot-validator
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: hono-middleware-repro@1.0.0
npm ERR! Found: valibot@0.30.0
npm ERR! node_modules/valibot
npm ERR!   valibot@"^0.30.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer valibot@"^0.13.1" from @hono/valibot-validator@0.2.2
npm ERR! node_modules/@hono/valibot-validator
npm ERR!   @hono/valibot-validator@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! /Users/ynkt/.npm/_logs/2024-04-06T18_55_32_176Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/ynkt/.npm/_logs/2024-04-06T18_55_32_176Z-debug-0.log
```

Semver like `>=0.13.1 <1` would be preferred so that users can have access to Valibot's new features, but I don't have time to investigate if there are incompatibilites that break `@hono/valibot-validator` between 0.13.1 and 0.30.0. Breaking changes might be OK because `@hono/valibot-validator` is pre-v1 as well.

I'd like to hear your opinion on this problem before working on a PR. What do you think?

yeah, trying to use valibot right now:

```$ npm add valibot @hono/valibot-validator
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: undefined@undefined
npm WARN Found: valibot@0.13.1
npm WARN node_modules/valibot
npm WARN   valibot@"*" from the root project
npm WARN   1 more (@hono/valibot-validator)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer valibot@"^0.13.1" from @hono/valibot-validator@0.2.2
npm WARN node_modules/@hono/valibot-validator
npm WARN   @hono/valibot-validator@"*" from the root project```

yeah, I was trying to use valibot

```$ npm add valibot @hono/valibot-validator
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: undefined@undefined
npm WARN Found: valibot@0.13.1
npm WARN node_modules/valibot
npm WARN   valibot@"*" from the root project
npm WARN   1 more (@hono/valibot-validator)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer valibot@"^0.13.1" from @hono/valibot-validator@0.2.2
npm WARN node_modules/@hono/valibot-validator
npm WARN   @hono/valibot-validator@"*" from the root project```

--- a/.changeset/eighty-doors-pull.md
+++ b/.changeset/eighty-doors-pull.md
@@ -1,0 +1,7 @@
+---
+'@hono/typebox-validator': patch
+'@hono/valibot-validator': patch
+'@hono/clerk-auth': patch
+---
+
+Update peer dependencies so that newer validators can be installed.

--- a/packages/clerk-auth/package.json
+++ b/packages/clerk-auth/package.json
@@ -38,7 +38,7 @@
   },
   "homepage": "https://github.com/honojs/middleware",
   "peerDependencies": {
-    "@clerk/backend": "0.30.*",
+    "@clerk/backend": ">=0.30.0 <1",
     "hono": ">=3.*"
   },
   "devDependencies": {

--- a/packages/typebox-validator/package.json
+++ b/packages/typebox-validator/package.json
@@ -27,7 +27,7 @@
   },
   "homepage": "https://github.com/honojs/middleware",
   "peerDependencies": {
-    "@sinclair/typebox": "^0.31.15",
+    "@sinclair/typebox": ">=0.31.15 <1",
     "hono": ">=3.9.0"
   },
   "devDependencies": {

--- a/packages/valibot-validator/package.json
+++ b/packages/valibot-validator/package.json
@@ -28,7 +28,7 @@
   "homepage": "https://github.com/honojs/middleware",
   "peerDependencies": {
     "hono": ">=3.9.0",
-    "valibot": "^0.13.1"
+    "valibot": ">=0.13.1 <1"
   },
   "devDependencies": {
     "hono": "^3.11.7",

